core/src/main/java/hudson/model/ComputerSet.java

```diff
@@ -109,8 +109,6 @@ public Set<LabelAtom> getAllLabels() {
             Set<LabelAtom> assignedNodeLabels = Label.parse(c.getNode().getLabelString());
             // + dynamic labels (some plugins use it)
             assignedNodeLabels.addAll(c.getNode().getDynamicLabels());
```

Comment:
this is wrong: you update assignedNodeLabels and the variable Set<LabelAtom> labels will be never changed.
Also we shall make the list unique and sorted.